src/Workspaces/Core/Portable/SymbolKey/SymbolKey.BodyLevelSymbolKey.cs

```diff
@@ -162,8 +164,42 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? fa
                     }
                 }
 
-                failureReason = $"({nameof(BodyLevelSymbolKey)} '{name}' not found)";
+                failureReason = $"({nameof(BodyLevelSymbolKey)} '{name}' not found -> {totalFailureReason})";
                 return default;
+
+                bool TryResolveLocation(Location loc, int index, out SymbolKeyResolution resolution, out string? reason)
+                {
+                    var resolutionOpt = reader.ResolveLocation(loc);
```

Comment:
Why the resolution/resolutionOpt? I only see a single use of 'resolution' and that's only to grab the symbol so worth simplifying it?